rllib/evaluation/tests/test_worker_set.py

```diff
@@ -0,0 +1,78 @@
+import gym
+import unittest
+
+import ray
+from ray.rllib.algorithms.algorithm_config import AlgorithmConfig
+from ray.rllib.evaluation.worker_set import WorkerSet
+from ray.rllib.examples.policy.random_policy import RandomPolicy
+from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID
+
+
+class TestWorkerSet(unittest.TestCase):
```

Comment:
In a follow-up PR, we should move the existing worker failure tests in here!

Reply:
sure.
actually, next PR, I will move all these async algorithm to use the WorkerSet directly.
so this will be tested quite extensively.